clayui.com/static/js/code-collapse.js

```diff
@@ -0,0 +1,35 @@
+if (document.body) {
+	document.body.addEventListener('click', event => {
+		const target = event.target;
+        const targetClassList = target.classList;
+        
+		if (
+			targetClassList.contains('btn-collapse') ||
+			targetClassList.contains('icon-collapse') ||
+			(target.tagName === 'use' &&
+				target.parentNode.classList.contains('icon-collapse'))
+		) {
+			const codeContainer = target.closest('.code-container');
```

Comment:
`.closest` isn't supported in IE, so we should either add a polyfil or use something else.

reference: https://developer.mozilla.org/en-US/docs/Web/API/Element/closest

Reply:
Ah yes, I'll have to work around that then. How dirty are we `allowed` to get with chaining and DOM queries?

Reply:
What exactly do you mean by "dirty"? I'd probably err on the side of not a ton of chaining, but it really depends on the situation. We don't have a hard and fast rule.

This is an example of closest we use in tooltip, https://github.com/liferay/clay/blob/master/packages/clay-tooltip/src/TooltipProvider.tsx#L80